File: benchmark/results/python-multilang/run-002/b-baseline/output/metaengine_demo/billing/value_objects/discount.py
"""Discount value object for the billing domain."""

from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class Discount:
    """Immutable line item describing a discount applied to an invoice."""

    id: str
    createdAt: datetime
    updatedAt: datetime
    name: str
    description: str

File: benchmark/results/java-multilang/run-003/a-mcp/output/billing/value_objects/Discount.java
package com.metaengine.demo.billing.value_objects;

import java.time.OffsetDateTime;

// Discount value object.

public class Discount {
    public String id;
    public OffsetDateTime createdAt;
    public OffsetDateTime updatedAt;
    public String name;
    public String description;

    public Discount(String id, OffsetDateTime createdAt, OffsetDateTime updatedAt, String name, String description) {
        this.id = id;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.name = name;
        this.description = description;
    }
}
